### PR TITLE
Fixes #259 Integrating Swagger UI with yaydoc

### LINKED ISSUE #259: Integrating Swagger UI with yaydoc
### This is a
- [ ] Bug report
- [x] Feature request

### Description

I did some research and also discussed with the team of susi_server. It was found that the only thing that need to exist in the gh-pages branch for Swagger to work is the Swagger-ui distribution with a change of API url in the index.html file.

The susi_server project already has the api URL set up at http://api.susi.ai/docs/swagger.json. We'll just have to integrate it.

### PR DESCRIPTION
**Pull request by imujjwal96**
## Description

Added 3 configuration variables in `.yaydoc.yml` to handle Swagger Docs
## Related Issue

Fixes #259 
## Motivation and Context

To enable documentation generation for projects like fossasia/susi_server that use Swagger to generate API documentation
## How Has This Been Tested?

Test at: http://yaydoc7.herokuapp.com

## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

File: generate.sh [hunk 1]
```diff
@@ -161,6 +161,16 @@ if [ $? -ne 0 ]; then
 fi
 print_log "Documentation Generated Successfully!\n"
 
+if [ -n "$APIDOCS_NAME" ] && [ -n "$APIDOCS_URL" ]; then
+  if [ "$APIDOCS_NAME" == "swagger" ]; then
+    cd ..
+    git clone -q https://github.com/swagger-api/swagger-ui.git swagger >/dev/null 2>&1
+    cd yaydoctemp/_build/html
+    cp -r ${BUILD_DIR}/../swagger/dist apidocs
```
Comment (pri22296) [inline generate.sh]: Instead of this clone directly and just move the directory to _build/html. The advantage is that you won't have to use the name `yaydoctemp` as BUILD_DIR is an abstraction for that.

Comment (imujjwal96) [inline generate.sh]: can cloning this in BUILD_DIR have any bad effect?

Comment (pri22296) [inline generate.sh]: I don't think so. what bad effects are we talking about?
It would be an isolated directory. After cloning just move `swagger/dist/.` to `_build/html` and then remove the swagger directory. I can't think of any side effects.

Comment (imujjwal96) [inline generate.sh]: @pri22296 updated it.

File: modules/scripts/config.py [hunk 2]
```diff
@@ -49,6 +49,10 @@ def _get_default_config(username, reponame):
             'publish': {'ghpages': {'docurl': None,},
                         'heroku': {'app_name': None,},
                        },
+            'apidocs': {'name': None,
```
Comment (pri22296) [inline modules/scripts/config.py]: Update the README

File: generate.sh [hunk 3]
```diff
@@ -161,6 +161,15 @@ if [ $? -ne 0 ]; then
 fi
 print_log "Documentation Generated Successfully!\n"
 
+if [ -n "$APIDOCS_NAME" ] && [ -n "$APIDOCS_URL" ]; then
+  if [ "$APIDOCS_NAME" == "swagger" ]; then
+    git clone -q https://github.com/swagger-api/swagger-ui.git swagger >/dev/null 2>&1
```
Comment (niranjan94) [inline generate.sh]: Instead of cloning their master, which can be unstable, get their latest stable release. 

`https://api.github.com/repos/swagger-api/swagger-ui/releases/latest` would give you a `zipball_url` parameter to get the zip of the latest release. Use that please.

Comment (sch00lb0y) [inline generate.sh]: since we cloning stable release. can we have to copy of `dist` folder in Yaydoc only?. it'll increase build speed. if this suggestion is useful. please consider

Comment (imujjwal96) [inline generate.sh]: A stable but the latest release. The latest release will change dynamically

**Comment by imujjwal96** [comment]
@niranjan94 Updated it!

File: generate.sh [hunk 4]
```diff
@@ -161,6 +161,17 @@ if [ $? -ne 0 ]; then
 fi
 print_log "Documentation Generated Successfully!\n"
 
+if [ -n "$APIDOCS_NAME" ] && [ -n "$APIDOCS_URL" ]; then
+  if [ "$APIDOCS_NAME" == "swagger" ]; then
+    STABLE_SWAGGER=$(curl -s https://api.github.com/repos/swagger-api/swagger-ui/releases/latest  | grep tarball_url | cut -d '"' -f 4) >/dev/null 2>&1
+    wget -O swagger.tar.gz ${STABLE_SWAGGER} >/dev/null 2>&1
+    mkdir swagger
+    tar -xf swagger.tar.gz --strip-components=1 -C swagger >/dev/null 2>&1
+    cp -r swagger/dist _build/html/apidocs
```
Comment (pri22296) [inline generate.sh]: Remove the swagger directory after copying

Comment (imujjwal96) [inline generate.sh]: We are removing the parent directory of yaydoctemp already! is this change required?

Comment (pri22296) [inline generate.sh]: Oh, cool. No need for it.
